objax/util/util.py

```diff
@@ -34,6 +36,38 @@ def __init__(self, *args, **kwargs):
         self.__dict__ = self
 
 
+class Renamer:
+    """Helper class for renaming string contents."""
+
+    def __init__(self,
+                 rules: Union[Dict[str, str], Sequence[Tuple[Pattern[str], str]], Callable[[str], str]],
+                 chain: Optional['Renamer'] = None):
+        """Create a renamer object.
+
+        Args:
+            rules: the replacement mapping.
+            chain: optionally, another renamer to call after this one completes.
+        """
+        self.chain = chain
+        if callable(rules):
+            self.subfn = rules
+        elif isinstance(rules, dict):
+            regex = re.compile('(%s)' % '|'.join(map(re.escape, rules.keys())))
+            self.subfn = functools.partial(regex.sub, lambda m: rules[m.string[m.start():m.end()]])
```

Comment:
I think `m.string[m.start():m.end()]` could be simplified as `m.group(0)`
see https://docs.python.org/3/library/re.html#re.Match.group